### ISSUE TITLE
[BUG] 'db_default_url' is not inherited by some DB-based modules

### ISSUE THREAD

**Issue by volga629-1** [OpenSIPS/opensips#2117]
Hello Everyone,
According doc of 3.1, but dispatcher unaware of default db url.

https://www.opensips.org/Documentation/Script-CoreParameters-3-1#toc13

1.12 db_default_url
The default DB URL to be used by modules if no per-module URL is given. Default is NULL (not defined)

Example of usage:

    db_default_url="mysql://opensips:opensipsrw@localhost/opensips"

```
May 21 07:00:39 aitossbc01 /usr/sbin/opensips[3584386]: ERROR:dispatcher:partition_init: [default] DB URL is not defined!
May 21 07:00:39 aitossbc01 /usr/sbin/opensips[3584386]: ERROR:core:init_mod: failed to initialize module dispatcher
May 21 07:00:39 aitossbc01 /usr/sbin/opensips[3584386]: ERROR:core:main: error while initializing modules
May 21 07:00:39 aitossbc01 /usr/sbin/opensips[3584386]: INFO:core:cleanup: cleanup

```

**Comment by liviuchircu** [comment]
hey, @volga629-1 !  This discussion actually started on the [users list](http://lists.opensips.org/pipermail/users/2020-April/042762.html), and the consensus was to proceed with inheriting `db_default_url` wherever it made sense to.  So let's use this thread in order to mark and fix all modules do not properly inherit that URL.

Will keep you posted on the progress.

**Comment by volga629-1** [comment]
Sure yes let continue in user list.

**Comment by liviuchircu** [comment]
Honestly, I prefer to continue here, because last time I didn't even remember to open up a ticket for it :)

**Comment by volga629-1** [comment]
sure :+1:

**Comment by volga629-1** [comment]
No resolution yet work in progress.

**Comment by volga629-1** [comment]
Work in progress

**Comment by liviuchircu** [comment]
hey @volga629-1, this is now addressed.  Here is a list of modules which have a new (hopefully improved) behavior with regards to `db_default_url` inheritance (fixed on `3.1` and `master` for now -- backports to `2.4` and `3.0` will follow in the upcoming months):

* b2b_entities
* b2b_logic
* dialplan
* dispatcher
* permissions
* proto_smpp

All other modules were carefully analyzed and they either correctly inherit the `db_default_url` already or they should _not_ inherit it as a default!  Example of the latter: `rtpengine`, `rtpproxy` or `tls_mgm` -- such modules may very well work without a DB URL thanks to static provisioning via `opensips.cfg` modparams, so their DB URL must be explicitly given.

**Comment by volga629-1** [comment]
@liviuchircu  Great thank you, I will test soon.
